3901 Update standard events display

liyiy opened the pull request:
## Changes

https://github.com/PostHog/posthog/issues/3901

Replace all frontend display instances of events prefixed with the `$`

- insights
<img width="740" alt="Screen Shot 2021-04-15 at 1 54 39 PM" src="https://user-images.githubusercontent.com/25164963/114939217-23115000-9e0e-11eb-9224-89ffc5d2c465.png">

- events 
<img width="341" alt="Screen Shot 2021-04-15 at 4 53 12 PM" src="https://user-images.githubusercontent.com/25164963/114939219-23115000-9e0e-11eb-8706-1dc02c5129e3.png">


## Checklist

- [x] All querysets/queries filter by Organization, by Team, and by User
- [x] Django backend tests
- [x] Jest frontend tests
- [x] Cypress end-to-end tests
- [x] Migrations are safe to run at scale (e.g. PostHog Cloud) – present proof if not obvious

liyiy committed:
Closes #3901 add PropertyKeyInfo to select box list

liyiy committed:
add more event properties

liyiy committed:
update standard events on insights dropdowns with PropertyKeyInfo

liyiy committed:
fix borked styling

liyiy committed:
add more events to the property key info list

liyiy committed:
update standard events with property key info for events tables

liyiy commented on frontend/src/lib/components/PropertyKeyInfo.js:
```diff
@@ -173,6 +173,52 @@ export const keyMapping = {
             description: 'The initial type of device that was used (first-touch).',
             examples: ['Mobile', 'Tablet', 'Desktop'],
         },
+        $pageview: {
```
I am happy to take suggestions for description or example updates to any of these properties

liyiy commented:
kind of strange that all of those typescript errors already existed before this PR but only fail now?

EDsCODE commented:
We did a gradual shift over from javascript to typescript and that includes adding a strict type check but only on files in a PR. You'll just have to address the typescript changes that the check picks up. I think the only weird typing issue will be with the `entityFilterLogicType` in the parameter. You should be able to change this to `BuiltLogic` from `kea` to get the typing to be happy

Note: There's a handful of javascript files left to convert and errors in some of the typescript files. We opted for a "fix it as we go" method so that it could be done on the spot while someone was fixing/adding features rather than having someone go through every single file. @samwinslow You may have encountered this but tagging you here so you're aware too

kpthatsme commented on frontend/src/lib/components/PropertyKeyInfo.js:
```diff
@@ -173,6 +173,52 @@ export const keyMapping = {
+            description: 'When a user leaves a page.',
+        },
+        $autocapture: {
+            label: 'Autocapture',
```
What are your thoughts on whether 'UI Interaction' would be more accurate for this one? Since most (all?) of these I believe are auto captured.

liyiy replied:
Do you mean to change the label of this to `UI Interaction` here?

liyiy replied:
For the label name specifically?

kpthatsme replied:
Yea for the label name

liyiy replied:
hmm would that provide more clarity than what the description for it already delivers?

kpthatsme replied:
Discussed offline– going to keep it as is so we're consistent roughly keeping the name consistent with the actual field.

liyiy committed:
typescript error fixes

EDsCODE commented:
#4003 fixes typescript issues in `ActionFilterDropdown` so you can rebase once we get that merged in a little while

liyiy committed:
resolve finicky typescript null object error

liyiy commented:
force pushed because i rebased 👍

liyiy committed:
Merge branch 'master' into 3901-update-standard-events-display

liyiy committed:
e2e updates

liyiy committed:
Merge branch '3901-update-standard-events-display' of https://github.com/PostHog/posthog into 3901-update-standard-events-display

liyiy committed:
update e2e

liyiy committed:
update e2e trends filter selector

liyiy committed:
Merge branch 'master' into 3901-update-standard-events-display

liyiy committed:
fix events filter bug

liyiy committed:
update break down button event name

liyiy commented:
Final updates:
- updated some e2e specs that broke 
- fixed an event filtering bug 
- updated the `break down by` button display 😸 
<img width="419" alt="Screen Shot 2021-04-19 at 12 22 20 PM" src="https://user-images.githubusercontent.com/25164963/115270403-4556e680-a10a-11eb-8356-5f718054d6e9.png">

liyiy committed:
e2e fix

liyiy committed:
Merge branch 'master' into 3901-update-standard-events-display

EDsCODE reviewed (commented):
lgtm! the small UI comment I would have is to adjust this so that the retention cohortizing event options can stay in line
<img width="331" alt="Screen Shot 2021-04-20 at 10 07 11 AM" src="https://user-images.githubusercontent.com/13127476/115410146-6417b480-a1c0-11eb-8883-364dd21bdf97.png">

liyiy commented:
> lgtm! the small UI comment I would have is to adjust this so that the retention cohortizing event options can stay in line
> <img alt="Screen Shot 2021-04-20 at 10 07 11 AM" width="331" src="https://user-images.githubusercontent.com/13127476/115410146-6417b480-a1c0-11eb-8883-364dd21bdf97.png">

What do you mean by stay in line? Like to have this dropdown be higher up?

<img width="486" alt="Screen Shot 2021-04-20 at 10 20 36 AM" src="https://user-images.githubusercontent.com/25164963/115412119-0edca280-a1c2-11eb-8496-431fe90c51a2.png">

EDsCODE commented:
Ahh I just mean as the boxes are aligned in a row currently so there's not too much vertical clutter when related items can be in a row
<img width="358" alt="Screen Shot 2021-04-20 at 10 34 41 AM" src="https://user-images.githubusercontent.com/13127476/115414543-1f8e1800-a1c4-11eb-91d5-c6ba2fb32699.png">

liyiy committed:
wrap in div and style fix

liyiy commented:
<img width="366" alt="Screen Shot 2021-04-20 at 12 14 33 PM" src="https://user-images.githubusercontent.com/25164963/115430500-6b47be00-a1d2-11eb-9e7a-be80b0536587.png">
<img width="335" alt="Screen Shot 2021-04-20 at 12 16 20 PM" src="https://user-images.githubusercontent.com/25164963/115430503-6be05480-a1d2-11eb-9289-124e583534c9.png">

EDsCODE reviewed (approved):
good to go
